extern/cfe/src/cfe.c

```diff
@@ -77,7 +77,9 @@ extern void init_soil_reservoir(cfe_model* cfe, double alpha_fc, double max_stor
             (upper_lim - lower_lim);
     // lateral flow parameters --------------------------------------------------------
     // 0.0 to deactiv. else =lateral_flow_linear_reservoir_constant;   // m per ts
-    cfe->soil_reservoir.coeff_secondary = get_K_lf_for_time_step(cfe);
+    //cfe->soil_reservoir.coeff_secondary = get_K_lf_for_time_step(cfe);
+    //Klf is a calibration parameter in the ngen code
+    cfe->soil_reservoir.coeff_secondary = 0.01;
```

Comment:
Even if `Klf` is not being calculated right currently with that function, hard-coded this value isn't correct.  Also, why that particular value?

Reply:
> For the second question, this is the first step in resolving the
> disagreement between BMI-CFE and the original CFE. Although not elegant, it
> produces consistent results with previously calibrated output thus
> demonstrating correctness of BMI-CFE code. I agree that this can be
> improved. There are possibly other places where parameters are hardcoded. I
> haven't gotten around to check all of them.

Wouldn't hard-coding the value like this cause the BMI CFE implementation to deviate from the original CFE if any other set of parameters is used?

Reply:
Actually, I see now in the latest version of Fred's code (that I have) where this is indeed hard-coded, but with contradicting comments (on about line 407):

`soil_reservoir.coeff_secondary=0.01;  // 0.0 to deactiv. else =lateral_flow_linear_reservoir_constant;   // m per h`

This has the look of something done as a temporary setting to work around or test something specific, then later be removed.  Perhaps we should ask Fred to clarify how this should actually operate.